server/sonar-server/src/main/java/org/sonar/server/computation/posttask/PostProjectAnalysisTasksExecutor.java

```diff
@@ -0,0 +1,191 @@
+    QualityGateHolder qualityGateHolder, QualityGateStatusHolder qualityGateStatusHolder,
+    @Nullable PostProjectAnalysisTask[] postProjectAnalysisTasks) {
+    this.analysisMetadataHolder = analysisMetadataHolder;
+    this.qualityGateHolder = qualityGateHolder;
+    this.qualityGateStatusHolder = qualityGateStatusHolder;
+    this.ceTask = ceTask;
+    this.postProjectAnalysisTasks = postProjectAnalysisTasks == null ? NO_POST_PROJECT_ANALYSIS_TASKS : postProjectAnalysisTasks;
+  }
+
+  @Override
+  public void finished(boolean allStepsExecuted) {
+    if (postProjectAnalysisTasks.length == 0) {
+      return;
+    }
+
+    ProjectAnalysis projectAnalysis = createProjectAnalysis(allStepsExecuted ? SUCCESS : FAILED);
+    for (PostProjectAnalysisTask postProjectAnalysisTask : postProjectAnalysisTasks) {
+      postProjectAnalysisTask.finished(projectAnalysis);
+    }
+  }
+
+  private ProjectAnalysis createProjectAnalysis(CeTask.Status status) {
+    return new ProjectAnalysis(
+      new CeTaskImpl(this.ceTask.getUuid(), status),
+      createProject(this.ceTask),
+      getAnalysisDate(),
+      status == SUCCESS ? createQualityGate(this.qualityGateHolder) : null);
+  }
+
+  private static Project createProject(org.sonar.ce.queue.CeTask ceTask) {
+    return new ProjectImpl(
+      ceTask.getComponentUuid(),
+      ceTask.getComponentKey(),
+      ceTask.getComponentName());
+  }
+
+  private Date getAnalysisDate() {
+    return new Date(this.analysisMetadataHolder.getAnalysisDate());
+  }
+
+  @CheckForNull
+  private QualityGateImpl createQualityGate(QualityGateHolder qualityGateHolder) {
+    Optional<org.sonar.server.computation.qualitygate.QualityGate> qualityGateOptional = qualityGateHolder.getQualityGate();
+    if (qualityGateOptional.isPresent()) {
+      org.sonar.server.computation.qualitygate.QualityGate qualityGate = qualityGateOptional.get();
+
+      return new QualityGateImpl(
+        qualityGate.getName(),
+        convert(qualityGateStatusHolder.getStatus()),
+        convert(qualityGate.getConditions(), qualityGateStatusHolder.getStatusPerConditions()));
+    }
+    return null;
+  }
+
+  private static QualityGate.Status convert(QualityGateStatus status) {
+    switch (status) {
+      case OK:
+        return QualityGate.Status.OK;
+      case WARN:
+        return QualityGate.Status.WARN;
+      case ERROR:
+        return QualityGate.Status.ERROR;
+      default:
+        throw new IllegalArgumentException(format(
+          "Unsupported value '%s' of QualityGateStatus can not be converted to QualityGate.Status",
+          status));
+    }
+  }
+
+  private static Collection<QualityGate.Condition> convert(Set<Condition> conditions, Map<Condition, ConditionStatus> statusPerConditions) {
+    return from(conditions)
+      .transform(new ConditionToCondition(statusPerConditions))
+      .toList();
+  }
+
+  private static class ProjectAnalysis implements PostProjectAnalysisTask.ProjectAnalysis {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2176)